test/system/tools/fds/CMakeLists.txt

```diff
@@ -24,21 +24,66 @@ if(PYTHONINTERP_FOUND)
         test_fast_discovery_closure
     )
 
+    # windows auxiliary script to fork test execution
+    set(PWS_LAUNCHER
+        ${CMAKE_CURRENT_SOURCE_DIR}/launcher.ps1
+    )
+
+    # Calculate environment
+    set(TEST_ENVIRONMENT 
+        "PATH=$ENV{PATH};$<TARGET_FILE_DIR:fastcdr>;$<TARGET_FILE_DIR:fastrtps>")
+
+    if(WIN32)
+
+        if(TARGET tinyxml2 OR TARGET tinyxml2::tinyxml2)
+            set(TEST_ENVIRONMENT 
+                "${TEST_ENVIRONMENT};$<TARGET_FILE_DIR:${TINYXML2_LIBRARY}>")
+        elseif(EXISTS TINYXML2_LIBRARY) 
+            get_filename_component(
+                TINYXML2_LIBRARY_DIR ${TINYXML2_LIBRARY} DIRECTORY)
+            set(TEST_ENVIRONMENT 
+                "${TEST_ENVIRONMENT};${TINYXML2_LIBRARY_DIR}")
+            unset(TINYXML2_LIBRARY_DIR)
```

Comment:
You may be unsetting an env variable that could be use somewhere else. Not sure what is the propouse, a comment will be helpful.

Reply:
I only change the environment for this specific test.
TEST_ENVIRONMENT has directory scope cannot be propagated upwards, thus, no side effects.